Video Player/config/jci/gui/apps/_videoplayer/js/videoplayer-v2.js

```diff
@@ -615,7 +644,7 @@ function myVideoPreviousRequest(){
 
 	if (previousVideoTrack === null)
 	{
-		previousVideoTrack = currentVideoTrack;
+		previousVideoTrack = recentlyPlayed.pop();
 	}
```

Comment:
There should be another check here for an empty array ie. first song in the recentlyPlayed list, maybe it could play currentVideoTrack--?  If you want to do this part its up to you 👍